aiohttp/connector.py

```diff
@@ -771,12 +778,18 @@ def _get_ssl_context(self, req):
             sslcontext = req.ssl
             if isinstance(sslcontext, ssl.SSLContext):
                 return sslcontext
+            if not (not isinstance(sslcontext, Fingerprint) or not (sslcontext.ssl_context is not None) or not isinstance(
+                    sslcontext.ssl_context, ssl.SSLContext)):
+                return sslcontext.ssl_context
             if sslcontext is not None:
                 # not verified or fingerprinted
                 return self._make_ssl_context(False)
             sslcontext = self._ssl
             if isinstance(sslcontext, ssl.SSLContext):
                 return sslcontext
+            if not (not isinstance(sslcontext, Fingerprint) or not (sslcontext.ssl_context is not None) or not isinstance(
```

Comment:
This is a duplicate check. Consider code reuse, so that it won't result in diverse copy-paste in future.